### PR TITLE
My budget App

### PR DESCRIPTION
**Pull request by emmanuelsteven**
### IN This PR We present the BudgetApp Production Build :
This application has undergone its development 👨‍💻 phase and completed successfully

Requirements
- Create and Setup a Rails Application
- Setup user Authentication with Devise gem.
- Create a landing page/Splash screen
- Create Categories model,controller and views as per requirements.
Create purchase(Transactions) model,controller and views as per requirements.
Create Unit and Integration tests for the application.
Follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)
ACHIEVEMENTS ⛰️
- Create and Setup a Rails Application✅
- Setup user Authentication with Devise gem.✅
- Create a landing page/Splash screen✅
- Create Categories model,controller and views as per requirements.✅
- Create purchase(Transactions) model,controller and views as per requirements.✅
- Create Unit and Integration tests for the application.✅
- Follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) ✅
Conclusion
I would love to hear 👂 from you, Our CR in terms of what can be better implemented 👨‍💻 for the site thank you! And we look forward to hearing from you soon. 🙇

### CODE REVIEW

File: app/views/splash/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (Lucas-Masaba) [inline app/views/splash/index.html.erb:1]: - [ ] Please make sure you follow the [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
  * Colors
  * Typography: font face, size and weight.
  * Layout: composition and space between elements.Check these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:

Please ensure that the splash screen looks similar to the original design as mentioned in the requirements.

### Your Implementation
image
### Required
image

File: app/views/categories/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  <% else %>
+  <p>Add a category</p>
+  <% end %>
+  </div>
+  </a>
+<% end %>
+<% end %>
+</div>
+<a href="<%= new_category_path %>" class='add-item'>
+<div class='inner'>
```
Comment (Lucas-Masaba) [inline app/views/categories/index.html.erb:32]: - [ ] Please make sure that the button spans across horizontally like in the original design.
### Your design
image
### Required
image

File: app/views/purchases/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        <p class='small-gap'><%= purchase.created_at.strftime('%e %b %Y') %></p>
+        <p class='small-gap'>$ <%= purchase.amount %></p>
+        
+      </div>
+    <% end %>
+    <% end %>
+  </div>
+</div>
+
+<a href='/categories/<%= @category.id %>/purchases/new' class='add-item'>
```
Comment (Lucas-Masaba) [inline app/views/purchases/index.html.erb:41]: - [ ] Please do the same thing with the button as I mentioned for the add a transaction button. Currently, yours looks like this:
image

File: app/views/purchases/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        <%= form.label :name %>
+        <%= form.text_field :name, autofocus: true, placeholder: 'Enter transaction name here...' %>
+      </div>
+
+      <div class='field'>
+        <%= form.label :amount %>
+        <%= form.number_field :amount, placeholder: 'Enter amount for the transaction...' %>
+      </div>
+      <%= hidden_field_tag 'category_id', @category.id %>
+
```
Comment (Lucas-Masaba) [inline app/views/purchases/new.html.erb:26]: - [ ] A user should be able to select a category on this form when creating a transaction.

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @emmanuelsteven  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Well written pull request message :writing_hand: 
- Professional README :scroll: 
- All written test passing :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Emmanuel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job implementing the design changes requested
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**